runtime/pallene_core.h

```diff
@@ -332,4 +332,15 @@ TValue *pallene_getshortstr(Table *t, TString *key, size_t * restrict pos)
         } \
     }
 
+/* If the table field size is smaller than LUAI_MAXSHORTLEN, use the optimized getStr Pallene implementation.
+ * Else, use Lua's default getStr method. */
+#define pallene_getfieldslot(field_len, tab, key, slot) \
+    if (field_len < LUAI_MAXSHORTLEN) { \
```

Comment:
Do you know if it would be possible to use a preprocessor `#if` instead of a regular `if`? That way the cache variable would only be created if it were needed.

Reply:
My first attempt was to implement it like this as you proposed, but according to [this](https://stackoverflow.com/a/10074611/8831148), one can't embed a preprocessor directive in another preprocessor directive.

It showed that error message when I tried doing it:
`error: '#' is not followed by a macro parameter`

Reply:
Doing it as proposed in the link would cause a compilation error since the variable `field_len` isn't defined at this point.
```c
#if field_len < LUAI_MAXSHORTLE
#define /* pallene_getshortstr implementation */
#else
#define /* luaH_getstr implementation */
#endif
```